From 8d41f377ad5de35aac935694a7d19427b14a0f31 Mon Sep 17 00:00:00 2001
From: Alex <leha-bot@users.noreply.github.com>
Date: Fri, 21 Aug 2020 20:12:58 +0300
Subject: [PATCH] Fix uniform initialization for WaveHeader[Pcm|IeeFloat] in
 C++20 mode

In C++17 and lower there was a nasty "feature" which allows to compile the structs with defined constructors using {} syntax. This "feature" was considered as defect in Standard and fixed in C++20.
---
 src/common_audio/wav_header.cc | 2 ++
 1 file changed, 2 insertions(+)

diff --git a/src/common_audio/wav_header.cc b/src/common_audio/wav_header.cc
index ce119f1..9c17ae3 100644
--- a/src/common_audio/wav_header.cc
+++ b/src/common_audio/wav_header.cc
@@ -80,6 +80,7 @@ const uint32_t kFmtIeeeFloatSubchunkSize =
 // read audio samples.
 #pragma pack(2)
 struct WavHeaderPcm {
+  WavHeaderPcm() = default;
   WavHeaderPcm(const WavHeaderPcm&) = default;
   WavHeaderPcm& operator=(const WavHeaderPcm&) = default;
   RiffHeader riff;
@@ -95,6 +96,7 @@ static_assert(sizeof(WavHeaderPcm) == kPcmWavHeaderSize,
 // WAV implementation.
 #pragma pack(2)
 struct WavHeaderIeeeFloat {
+  WavHeaderIeeeFloat() = default;
   WavHeaderIeeeFloat(const WavHeaderIeeeFloat&) = default;
   WavHeaderIeeeFloat& operator=(const WavHeaderIeeeFloat&) = default;
   RiffHeader riff;
-- 
2.30.0

